### ISSUE TITLE
Issues using HTC Vive in #177

### ISSUE THREAD

**Issue by De-Panther** [Unity-Technologies/EditorVR#178]
I installed Unity 5.6.0xEditorVR-p3 (64-bit)
And tested it using pull #177 

When opening the EditorVR window, I see the controllers without the Unity menu icon, and I get these errors :
1)
ArgumentOutOfRangeException: startIndex + length > this.length
Parameter name: length
System.String.Substring (Int32 startIndex, Int32 length) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System/String.cs:356)
UnityEditor.Experimental.EditorVR.UI.GradientButton.SetContent (System.String displayedText) (at Assets/EditorVR/Scripts/UI/GradientButton.cs:493)
UnityEditor.Experimental.EditorVR.UI.GradientButton.Awake () (at Assets/EditorVR/Scripts/UI/GradientButton.cs:189)
UnityEditor.Experimental.EditorVR.Input.ViveInputToEvents:Update()

m_Text.text = displayedText.Substring(0, 2);
But the base text on the prefab is only 1 char.

2)
Array index (1) is out of bounds (size=0)
In UnityEditor.Experimental.EditorVR.Menus.MainMenu:Start()

3)
NullReferenceException: Object reference not set to an instance of an object
UnityEditor.Experimental.EditorVR.Proxies.DefaultProxyRay.get_pointerLength () (at Assets/EditorVR/Scripts/Proxies/DefaultProxyRay.cs:70)
UnityEditor.Experimental.EditorVR.Core.EditorVR+DirectSelection.GetPointerLength (UnityEngine.Transform rayOrigin) (at Assets/EditorVR/Scripts/Core/EditorVR.DirectSelection.cs:66)
UnityEditor.Experimental.EditorVR.IGetPointerLengthMethods.GetPointerLength (IGetPointerLength obj, UnityEngine.Transform rayOrigin) (at Assets/EditorVR/Scripts/Core/Interfaces/IGetPointerLength.cs:17)
UnityEditor.Experimental.EditorVR.Modules.MultipleRayInputModule.ProcessInput (UnityEngine.InputNew.ActionMapInput input, UnityEditor.Experimental.EditorVR.ConsumeControlDelegate consumeControl) (at Assets/EditorVR/Scripts/Modules/MultipleRayInputModule/MultipleRayInputModule.cs:132)
UnityEditor.Experimental.EditorVR.Core.EditorVR.ProcessInput (System.Collections.Generic.HashSet`1 processedInputs, UnityEditor.Experimental.EditorVR.ConsumeControlDelegate consumeControl) (at Assets/EditorVR/Scripts/Core/EditorVR.cs:322)
UnityEditor.Experimental.EditorVR.Modules.DeviceInputModule.ProcessInput () (at Assets/EditorVR/Scripts/Modules/DeviceInputModule.cs:90)
UnityEditor.Experimental.EditorVR.Core.EditorVR.Update () (at Assets/EditorVR/Scripts/Core/EditorVR.cs:307)

**Comment by sarahhockman** [comment]
@De-Panther I also experienced this issue. I found it to be related to the pull command.

I experienced this when using: 
`git lfs clone --recursive -b staging http://github.com/Unity-Technologies/EditorVR`

After using that command, there is an issue with `git lfs`, which results in `git lfs error 301`:
`Git LFS: (0 of 131 files) 0 B / 14.81 MB
Invalid status for POST http://github.com/Unity-Technologies/EditorVR.git/info/lfs/objects/batch: 301
Invalid status for POST http://github.com/Unity-Technologies/EditorVR.git/info/lfs/objects/batch: 301
Entering 'libs/UnityOctree'
Entering 'libs/VRLineRenderer'
Entering 'libs/input-prototype`

In short, the result is being unable to see the Menus or use the controllers in EVR.

To resolve this, do another pull, and use **https://github.com/Unity-Technologies/EditorVR** in your command line, as opposed to http://github.com/Unity-Technologies/EditorVR
(make sure to include 'https' instead of 'http').

`git lfs clone --recursive -b staging https://github.com/Unity-Technologies/EditorVR`

**Comment by mtschoen-unity** [comment]
Hm. This is strange, indeed. I cannot confirm it on my end by just checking out to this branch.

**Comment by amirebrahimi** [comment]
@sarahhockman, I just tried pulling with only http and had the same problem. I wonder if GH LFS requires it now. I'll update the docs, regardless.

**Comment by amirebrahimi** [comment]
I couldn't repro this either with a fresh pull (using HTTPS) into a new project with Vive.

**Comment by sarahhockman** [comment]
_Also to note:_ 
Using Oculus, I noticed that using 'http' instead of 'https' in the command results in the controllers becoming entirely invisible in EVR.

But, when consistently using a pull command with 'https', I experience no issues. (With both Vive + Oculus)

@amirebrahimi I believe you're right, and GH LFS requires it. Good thing to make note of.

**Comment by De-Panther** [comment]
I'm downloading zip files from Github instead of using git to test the staging brunch.(And also downloading zip files of the submodules)
Maybe when working that way there are issues with LFS.
I guess I'll just install Git LFS and configure it for the next time that I'll want to check the staging brunch.
